src/Illuminate/Database/Eloquent/Builder.php

```diff
@@ -70,6 +77,55 @@ public function __construct(QueryBuilder $query)
     }
 
     /**
+     * Register a new global scope.
+     *
+     * @param  \Illuminate\Database\Eloquent\ScopeInterface|\Closure  $scope
+     * @param  string  $identifier
+     * @return $this
+     */
+    public function applyGlobalScope($scope, $identifier)
+    {
+        $this->scopes[$identifier] = $scope;
```

Comment:
If ``$identifier`` is not provided a random string (``uniqid``) is generated instead in the Model class. Identifier is really only needed if you wish to remove anonymous scope.

But yeah I was also thinking about making it the first argument. I'm just not sure if that isn't too "magical" because the method signature wouldn't really match the arguments.